egui/src/widgets/slider.rs

```diff
@@ -247,13 +272,20 @@ impl<'a> Slider<'a> {
     }
 
     fn set_value(&mut self, mut value: f64) {
-        if self.clamp_to_range {
-            let start = *self.range.start();
-            let end = *self.range.end();
-            value = value.clamp(start.min(end), start.max(end));
-        }
-        if let Some(max_decimals) = self.max_decimals {
-            value = emath::round_to_decimals(value, max_decimals);
+        if let Some(step) = self.step {
+            let remainer = value % step;
+            if remainer != 0.0 {
+                value -= remainer;
+            };
+        } else {
+            if self.clamp_to_range {
```

Comment:
We should be able to both clamp to range and have stepping (as long as the step size is smaller than the range)

Reply:
I've moved the "stepping code" to the bottom and now it even works better :)
Looks like clamping does not interfere with steps, and because filtering is performed after everything, we can't get a value outside step interval.
In the worst case (clamping and step value out of range), you can't change a value from zero.